File: ru.kmorozov.gbd.core.logger/src/main/java/ru/kmorozov/gbd/logger/listeners/IEventListener.java
package ru.kmorozov.gbd.logger.listeners;

import ru.kmorozov.gbd.logger.events.BaseEvent;

/**
 * Created by km on 15.12.2015.
 */
public interface IEventListener {

    void receiveEvent(BaseEvent event);

    boolean eventMatched(BaseEvent event);
}

File: ru.kmorozov.gbd.core.logger/src/main/java/ru/kmorozov/gbd/logger/listeners/IEventListener.kt
package ru.kmorozov.gbd.logger.listeners

import ru.kmorozov.gbd.logger.events.LogEvent

/**
 * Created by km on 15.12.2015.
 */
interface IEventListener {

    fun receiveEvent(event: LogEvent)

    fun eventMatched(event: LogEvent): Boolean
}
